docs/solutions/workflow-issues/tokiocompat-deduplication-2026-05-07.md

```
@@ -0,0 +1,189 @@
+  - tokio
+  - futures-io
+plan_ref: issue-72-tokiocompat-dedup
+---
+
+## Problem
+
+`TokioCompat<T>` adapter (bridging tokio's `AsyncRead`/`AsyncWrite` to `futures_io` traits) existed in three separate locations across `harnx-acp` and `harnx-acp-server` crates, causing maintenance burden and potential drift.
+
+## Symptoms
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)